algorithms/src/snark/marlin/ahp/verifier/messages.rs

```diff
@@ -90,15 +97,16 @@ impl<F: PrimeField> QuerySet<F> {
     /// `(polynomial_label, (query_label, query))`.
     pub fn to_set(&self) -> crate::polycommit::sonic_pc::QuerySet<'_, F> {
         let mut query_set = crate::polycommit::sonic_pc::QuerySet::new();
-        for i in 0..self.batch_size {
-            query_set.insert((witness_label("z_b", i), self.z_b_query.clone()));
+        for (circuit_id, &batch_size) in self.batch_sizes.iter() {
+            for i in 0..batch_size {
+                query_set.insert((witness_label(&circuit_id, "z_b", i), self.z_b_query.clone()));
+                query_set.insert((witness_label(&circuit_id, "g_a", i), self.g_a_query.clone()));
+                query_set.insert((witness_label(&circuit_id, "g_b", i), self.g_b_query.clone()));
+                query_set.insert((witness_label(&circuit_id, "g_c", i), self.g_c_query.clone()));
```

Comment:
There's only one `g_m` per circuit, no? So the `g_a`, `g_b`, `g_c` computations should happen outside the inner loop, and should not include `i`

Reply:
Indeed, thx, fixed